### PR TITLE
CFY-2993 list response impl

### PR DESCRIPTION
**Pull request by adamlavie**

### CODE REVIEW

File: rest-service/manager_rest/test/test_list_pagination.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        self._put_n_deployments(id_prefix='test', number_of_deployments=2)
+        last = {'_offset': 1, '_size': 2}
+        response = self._put_deployment_modification(
+            deployment_id='test0_deployment')
+        self._mark_deployment_modification_finished(
+            modification_id=response['id'])
+        self._put_deployment_modification(deployment_id='test1_deployment')
+        response = self.client.deployment_modifications.list(**last)
+        self.assert_pagination_result(last, response, 2)
+
```
Comment (adamlavie) [inline rest-service/manager_rest/test/test_list_pagination.py]: implement

File: tests/workflow_tests/test_rest_service_pagination.py [hunk 2]
```diff
@@ -80,11 +81,13 @@ def test_plugins_pagination(self):
...
     def _test_pagination(self, list_func):
-        all_results = list_func()
+        all_results = list_func().items
         num_all = len(all_results)
         # sanity
         self.assertGreaterEqual(num_all, 10)
         for offset in range(num_all + 1):
             for size in range(num_all + 1):
                 response = list_func(_offset=offset, _size=size)
-                self.assertEqual(response, all_results[offset:offset+size])
```
Comment (adamlavie) [inline tests/workflow_tests/test_rest_service_pagination.py]: offset and page_size

File: rest-service/manager_rest/manager_elasticsearch.py [hunk 3]
```diff
@@ -128,3 +128,12 @@ def search(index=None, doc_type=None, body=None):
         """
         es = ManagerElasticsearch.get_connection()
         return es.search(index=index, doc_type=doc_type, body=body)
+
+    @staticmethod
+    def build_list_result_metadata(query, search_result):
+
+        pagination = {'total': search_result['hits']['total'],
```
Comment (adamlavie) [inline rest-service/manager_rest/manager_elasticsearch.py]: change page_size to size

File: rest-service/manager_rest/resources.py [hunk 4]
```diff
@@ -129,6 +130,11 @@ def wrapper(*args, **kwargs):
```
Comment (adamlavie) [inline rest-service/manager_rest/resources.py]: should marshal be moved to endpoint module or wherever

File: rest-service/manager_rest/resources.py [hunk 5]
```diff
@@ -149,6 +155,11 @@ def wrap_with_response_object(self, data):
         raise RuntimeError('Unexpected response data type {0}'.format(
             type(data)))
 
+    def wrap_list_with_response_object(self, data, fields_to_include):
```
Comment (adamlavie) [inline rest-service/manager_rest/resources.py]: move inline

File: rest-service/manager_rest/resources_v2.py [hunk 6]
```diff
@@ -894,8 +912,19 @@ def _build_query(filters=None, pagination=None, sort=None,
                                range_filters=range_filters)
 
     @staticmethod
+    def list_events(query, include=None):
+        result = Events._search(query, include=include)
```
Comment (adamlavie) [inline rest-service/manager_rest/resources_v2.py]: Events == self

File: rest-service/manager_rest/resources_v2.py [hunk 7]
```diff
@@ -920,22 +950,8 @@ def get(self, _include=None, filters=None,
...
-                'pagination': {
-                    'total': total_events,
-                    'offset': pagination.get('offset', 0),
-                    'size': pagination.get('page_size', len(events))
-                }
-            },
-            'items': events
-        }
-        return response
+        search_result = Events.list_events(query, include=_include)
```
Comment (adamlavie) [inline rest-service/manager_rest/resources_v2.py]: return single line

File: rest-service/manager_rest/test/base_test.py [hunk 8]
```diff
@@ -46,6 +46,10 @@
 LATEST_API_VERSION = 2  # to be used by max_client_version test attribute
```
Comment (adamlavie) [inline rest-service/manager_rest/test/base_test.py]: leave as constant

File: rest-service/manager_rest/test/test_events_v2.py [hunk 9]
```diff
@@ -121,4 +125,8 @@ def _get_expected_query(self):
             'size': 5,
             'from': 3
         }
-        return expected_query
```
Comment (adamlavie) [inline rest-service/manager_rest/test/test_events_v2.py]: sort other list as well

File: rest-service/manager_rest/test/test_list_pagination.py [hunk 10]
```diff
@@ -0,0 +1,71 @@
+#########
```
Comment (adamlavie) [inline rest-service/manager_rest/test/test_list_pagination.py]: talk to chen about this class in context of snapshots

File: rest-service/manager_rest/test/test_modify.py [hunk 11]
```diff
@@ -135,19 +141,32 @@ def node_assertions(num, deploy_num, planned_num):
...
-        self.assertEqual([], self.client.deployment_modifications.list(
-            deployment_id='i_really_should_not_exist'))
-        self.assertEqual(modification.node_instances.before_modification,
-                         before_modification)
-        self.assertEqual(modification.node_instances.before_rollback,
-                         expected_before_end_func(before_end))
-
-        self.assertEqual(after_end,
-                         expected_after_end_func(before_modification,
-                                                 before_end))
```
Comment (adamlavie) [inline rest-service/manager_rest/test/test_modify.py]: refactor as best as possible

File: rest-service/manager_rest/test/test_events_v2.py [hunk 12]
```diff
@@ -121,4 +125,10 @@ def _get_expected_query(self):
...
+        return self.sort_query_list(expected_query)
+
+    def sort_query_list(self, query):
+        query['query']['filtered']['filter']['bool']['must'].sort()
+        query['query']['filtered']['filter']['bool']['must'][
```
Comment (dankilman) [inline rest-service/manager_rest/test/test_events_v2.py]: is it guaranteed to always be the 3rd item?

Comment (adamlavie) [inline rest-service/manager_rest/test/test_events_v2.py]: if I do it after the initial sorting is done, then yea

**Pull request merged**